src/caf/base/data_structures.py

```diff
@@ -939,7 +944,10 @@ def _generic_dunder(
 
     def __len__(self):
         """Return the length of a DVector, defined as number of cells."""
```

Comment:
Why are we defining length as number of cells instead of number of rows to be inline with pandas and numpy? If we need number of cells I think this might be better as a size property